src/Model/Resolver/Product/PriceRange.php

```diff
@@ -135,7 +137,53 @@ protected function formatPrice(float $regularPrice, float $finalPrice, StoreInte
                 'value' => $finalPrice,
                 'currency' => $store->getCurrentCurrencyCode()
             ],
-            'discount' => $this->discount->getDiscountByDifference($regularPrice, $finalPrice),
+            'discount' => $discount,
         ];
     }
+
+    /**
+     * Calculates correct discount amount
+     * - Bundle items can contain $regularPrice and $finalFrice from two different
+     * - product instances, thus we are intersted in BE set special price procentage.
+     *
+     * @param Product $product
+     * @param float $regularPrice
+     * @param float $finalPrice
+     * @return array
+     */
+    protected function calculateDiscount(Product $product, float $regularPrice, float $finalPrice) : array
+    {
+        if ($product->getTypeId() != 'bundle') {
```

Comment:
Prefer strict comparison (`!==`) when it is possible

Reply:
Added strict comparsion